server/routes/productRoute.js

```diff
@@ -54,19 +81,86 @@ router.post("/search", async (req, res) => {
   }
 });
 
-router.post("/addProduct", async (req, res) => {
-  if (!req.session.AuthCookie) {
-    res.status(401).redirect("/");
-  } else {
-    //check what all is required after making the front end form
-    const {
-      product_Name,
+router.post("/addProduct", upload.single("logo"), async (req, res) => {
```

Comment:
can you use authmiddleware here ?

Reply:
Oh yeah sorry I had removed it to make things easier to test. I'll add it all back and push it soon